src/Features/LanguageServer/Protocol/Handler/RequestContextFactory.cs

```diff
@@ -32,7 +33,8 @@ public Task<RequestContext> CreateRequestContextAsync<TRequestParam>(IQueueItem<
         }
 
         TextDocumentIdentifier? textDocumentIdentifier;
-        var textDocumentIdentifierHandler = queueItem.MethodHandler as ITextDocumentIdentifierHandler;
+        var handler = queueItem.MethodHandler ?? throw new InvalidOperationException($"No handler available for method: {queueItem.MethodName}");
```

Comment:
would make the property non-nullable.  the impl can also throw if called before the value is set.